code/sorting/src/merge_sort/merge_sort.kt

```diff
@@ -0,0 +1,33 @@
+import java.util.*
+fun <T : Comparable<T>>mergesort(items : MutableList<T>) : MutableList<T> {
+    if (items.isEmpty()) {
+        return items
+    }
+     fun merge(left : MutableList<T>, right : MutableList<T>) : MutableList<T> {
+        var merged : MutableList<T> = arrayListOf<T>()
+        while(!left.isEmpty() && !right.isEmpty()) {
+            val temp : T
+            if (left.first() < right.first()) {
+                temp = left.removeAt(0)
+            }
+            else {
+                temp = right.removeAt(0)
+            }
+            merged.add(temp)
+        }
+        if (!left.isEmpty()) merged.addAll(left)
+        if (!right.isEmpty()) merged.addAll(right)
+         return merged
```

Comment:
Indentation.

Reply:
@arnavb please check your mail.

Reply:
@Awani-Shukla I will unfortunately be unable to accept your offer, as I have many academic demands (I am a mere high school student). Sorry!

Reply:
no problem.